src/libs/actions/TaxRate.ts

```diff
@@ -486,6 +487,33 @@ function renamePolicyTax(policyID: string, taxID: string, newName: string) {
 function setPolicyTaxCode(policyID: string, oldTaxCode: string, newTaxCode: string) {
     const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
     const originalTaxRate = {...policy?.taxRates?.taxes[oldTaxCode]};
+    const customUnits = Object.values(policy?.customUnits ?? {});
+    const optimisticCustomUnit = {
+        customUnits: {
+            ...customUnits.reduce((units, customUnit) => {
+                // eslint-disable-next-line no-param-reassign
+                units[customUnit.customUnitID] = {
+                    rates: {
+                        ...Object.keys(customUnit.rates).reduce((rates, rateID) => {
+                            if (customUnit.rates[rateID].attributes?.taxRateExternalID === oldTaxCode) {
+                                // eslint-disable-next-line no-param-reassign
+                                rates[rateID] = {
+                                    attributes: {
```

Comment:
also do we need to spread the old rate here? to make sure we don't remove everything but `attributes`?

Reply:
oh yep